### PR TITLE
assert volume tags

### PR DESCRIPTION
**Pull request by ppapishe**
### SUMMARY
* Adding a new method to Assert the volume tags

### DEVELOPER IMPACT
* No Breaking changes as this s a new method

### Contribution Checklist
* [x] All changes have been reflected in comparable unit test changes or additions.
* [x] Any interactions with third party clients are done via interface types rather than direct interactions.
* [x] All new functions follow the required naming standard.
* [x] All new functions follow the required signature standards.

### PIC
Embed something funny here

### CODE REVIEW

File: pkg/aws/ec2.go [hunk 1]
```diff
@@ -75,6 +75,16 @@ type AssertEC2TagValueInput struct {
...
 }
 
+// AssertEC2VolumeTagValueInput is used as an input to the AssertEC2VolumeTagValue method.
+type AssertEC2VolumeTagValueInput struct {
+	// The name of the tag to assert exists.
+	TagName string
+	// The value of the tag to assert.
+	Value string
+	// The Volume ID that the method will assert has a tag with the specified tag name and the specified value.
+	VolumeId string
```
Comment (yardbirdsax) [inline pkg/aws/ec2.go:85]: **Nit:** Idiomatic naming says "ID" should be capitalized.
```suggestion
	VolumeID string
```

File: pkg/aws/ec2.go [hunk 2]
```diff
@@ -231,6 +241,18 @@ func AssertEC2TagValue(t *testing.T, ctx context.Context, client EC2Client, inpu
...
+func AssertEC2VolumeTagValue(t *testing.T, ctx context.Context, client EC2Client, input AssertEC2VolumeTagValueInput) {
+	volume, _ := getEC2VolumeByVolumeIDE(ctx, client, input.VolumeId)
+	for _, tag := range volume.Tags {
+		tagKey := *tag.Key
+		tagValue := *tag.Value
+		if tagKey == input.TagName {
+			assert.Equal(t, input.Value, tagValue, "Tag with key '%s' does not match expected value.", tagKey)
+		}
+	}
+}
```
Comment (yardbirdsax) [inline pkg/aws/ec2.go:254]: **Suggestion:** If the volume does not have the specified tag at all (regardless of the value), the test should fail. (Unit tests should cover this scenario as well.)

**Review by yardbirdsax — CHANGES_REQUESTED** [review]
Thanks for the PR @ppapishe ! A few small comments.

File: pkg/aws/ec2.go (hunk 1 above)
Comment (ppapishe) [reply pkg/aws/ec2.go]: done

File: pkg/aws/ec2.go (hunk 2 above)
Comment (ppapishe) [reply pkg/aws/ec2.go]: added .

**Review by yardbirdsax — APPROVED** [review]
Nice work @ppapishe ! LGTM

**Pull request merged**